components/org.wso2.carbon.identity.media.core/src/main/java/org/wso2/carbon/identity/media/core/StorageSystem.java

```diff
@@ -31,10 +32,19 @@ String addMedia(List<InputStream> inputStreams, MediaMetadata mediaMetadata, Str
 
     DataContent getFile(String id, String tenantDomain, String type) throws StorageSystemException;
 
-    boolean evaluateSecurity(String accessLevel, String id, String type, String tenantDomain,
-                             String[] oauth2AllowedScopes) throws StorageSystemException;
+    boolean evaluateDownloadSecurityForPublicMedia(String id, String type, String tenantDomain) throws
+            StorageSystemException;
 
-    void deleteFile(String id, String type, String tenantDomain) throws StorageSystemException;
+    boolean evaluateDownloadSecurityForProtectedMedia(String id, String type, String tenantDomain) throws
+            StorageSystemException;
+
+    boolean evaluateMediaManagementSecurityForEndUser(String id, String type, String tenantDomain) throws
+            StorageSystemException;
+
+    boolean isMediaDeleted(String id, String type, String tenantDomain) throws StorageSystemException;
```

Comment:
Shall we change the method names to start with is since this returning a boolean?